site/app/controllers/forum/ForumController.php

```diff
@@ -705,180 +709,177 @@ public function markPostUnread(): JsonResponse {
-                if ($type == "Post") {
-                    $post_content = $_POST["thread_post_content"];
-                    $subject = "Post Edited: " . Notification::textShortner($post_content);
-                    $content = "A message was edited in:\n" . $full_course_name . "\n\nThread Title: " . $thread_title . "\n\nEdited Post: \n\n" . $post_content;
-                }
-                elseif ($type == "Thread and Post") {
-                    $post_content = $_POST["thread_post_content"];
-                    $subject = "Thread Edited: " . Notification::textShortner($thread_title);
-                    $content = "A thread was edited in:\n" . $full_course_name . "\n\nEdited Thread: " . $thread_title . "\n\nEdited Post: \n\n" . $post_content;
-                }
-                else {
-                    $subject = "Thread Edited: " . Notification::textShortner($thread_title);
-                    $content = "A thread was edited in:\n" . $full_course_name . "\n\nEdited Thread: " . $thread_title;
-                }
-
-                $event = ['component' => 'forum', 'metadata' => $metadata, 'content' => $content, 'subject' => $subject, 'recipient' => $post_author_id, 'preference' => 'all_modifications_forum'];
-                $this->core->getNotificationFactory()->onPostModified($event);
-            }
-            if ($isError) {
-                return $this->core->getOutput()->renderJsonFail($messageString);
-            }
-            if ($type === 'Post') {
-                $posts = $this->core->getQueries()->getPostsInThreads([$thread_id]);
-                $order_and_replies = self::getPostsOrderAndReplies($posts, $thread_id);
-                $order_array = $order_and_replies[0];
-                $reply_level_array = $order_and_replies[1];
-
-                $place = array_search($post["id"], $order_array, true);
-                $reply_level = $reply_level_array[$place];
-                $post_box_id = 1;
+    }
 
-                $this->sendSocketMessage([
-                    'type' => 'edit_post',
-                    'thread_id' => $thread_id,
-                    'post_id' => $post_id,
-                    'reply_level' => $reply_level,
-                    'post_box_id' => $post_box_id,
-                ]);
-            }
-            elseif ($type === 'Thread and Post') {
-                $this->sendSocketMessage([
-                    'type' => 'edit_thread',
-                    'thread_id' => $thread_id,
-                    'post_id' => $post_id,
-                    'reply_level' => 1,
-                    'post_box_id' => 1,
-                ]);
-            }
-            return $this->core->getOutput()->renderJsonSuccess(['type' => $type]);
+    private function recursiveDeletePost(Post $post, Thread $thread): void {
+        if ($thread->getFirstPost()->getId() === $post->getId()) {
+            $thread->setDeleted(true);
+        }
```

Comment:
I would pull this out of this function as it only would get used the first iteration of the recurse and the function is only called on time.